src/server/ns_turn_server.c

```diff
@@ -4681,80 +4681,79 @@ static void peer_input_handler(ioa_socket_handle s, int event_type, ioa_net_data
 
-        ioa_network_buffer_header_init(nbh);
+    ioa_network_buffer_header_init(nbh);
 
-        SOCKET_TYPE st = get_ioa_socket_type(ss->client_socket);
-        int do_padding = is_stream_socket(st);
+    SOCKET_TYPE st = get_ioa_socket_type(ss->client_socket);
+    int do_padding = is_stream_socket(st);
 
-        stun_init_channel_message_str(chnum, ioa_network_buffer_data(nbh), &len, len, do_padding);
-        ioa_network_buffer_set_size(nbh, len);
-        in_buffer->nbh = NULL;
-        if (eve(server->verbose)) {
-          TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "%s: send channel 0x%x\n", __FUNCTION__, (int)(chnum));
-        }
-      } else {
+    stun_init_channel_message_str(chnum, ioa_network_buffer_data(nbh), &len, len, do_padding);
+    ioa_network_buffer_set_size(nbh, len);
+    in_buffer->nbh = NULL;
+    if (eve(server->verbose)) {
+      TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "%s: send channel 0x%x\n", __FUNCTION__, (int)(chnum));
+    }
+  } else {
 
-        size_t len = 0;
+    size_t len = 0;
 
-        nbh = ioa_network_buffer_allocate(server->e);
-        stun_init_indication_str(STUN_METHOD_DATA, ioa_network_buffer_data(nbh), &len);
-        stun_attr_add_str(ioa_network_buffer_data(nbh), &len, STUN_ATTRIBUTE_DATA,
-                          ioa_network_buffer_data(in_buffer->nbh), (size_t)ilen);
-        stun_attr_add_addr_str(ioa_network_buffer_data(nbh), &len, STUN_ATTRIBUTE_XOR_PEER_ADDRESS,
-                               &(in_buffer->src_addr));
-        ioa_network_buffer_set_size(nbh, len);
-
-        if (!(*server->no_software_attribute)) {
-          const uint8_t *field = (const uint8_t *)get_version(server);
-          size_t fsz = strlen(get_version(server));
-          size_t len = ioa_network_buffer_get_size(nbh);
-          stun_attr_add_str(ioa_network_buffer_data(nbh), &len, STUN_ATTRIBUTE_SOFTWARE, field, fsz);
-          ioa_network_buffer_set_size(nbh, len);
-        }
+    nbh = ioa_network_buffer_allocate(server->e);
+    stun_init_indication_str(STUN_METHOD_DATA, ioa_network_buffer_data(nbh), &len);
+    stun_attr_add_str(ioa_network_buffer_data(nbh), &len, STUN_ATTRIBUTE_DATA, ioa_network_buffer_data(in_buffer->nbh),
+                      (size_t)ilen);
+    stun_attr_add_addr_str(ioa_network_buffer_data(nbh), &len, STUN_ATTRIBUTE_XOR_PEER_ADDRESS, &(in_buffer->src_addr));
+    ioa_network_buffer_set_size(nbh, len);
 
-        if ((server->fingerprint) || ss->enforce_fingerprints) {
-          size_t len = ioa_network_buffer_get_size(nbh);
-          stun_attr_add_fingerprint_str(ioa_network_buffer_data(nbh), &len);
-          ioa_network_buffer_set_size(nbh, len);
-        }
-      }
-      if (eve(server->verbose)) {
-        uint16_t *t = (uint16_t *)ioa_network_buffer_data(nbh);
-        TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "Send data: 0x%x\n", (int)(nswap16(t[0])));
-      }
+    if (!(*server->no_software_attribute)) {
+      const uint8_t *field = (const uint8_t *)get_version(server);
```

Comment:
minor: consider east-const instead of const-west.

`uint8_t const* const field = ....` is much more clear than `const uint8_t * const field = ...`

Reply:
we use const uint8_t* everywhere in the code so we should keep it that way imo